src/pages/[platform]/build-a-backend/server-side-rendering/nuxt/index.mdx

```diff
@@ -0,0 +1,626 @@
+import config from '~/amplify_outputs.json';
+
+const amplifyConfig = parseAmplifyConfig(config);
+
+const createCookieStorageAdapter = (
+  event: H3Event<EventHandlerRequest>
+): CookieStorage.Adapter => {
+  // `parseCookies`, `setCookie` and `deleteCookie` are Nuxt provided functions
+  const readOnlyCookies = parseCookies(event);
+
+  return {
+    get(name) {
+      if (readOnlyCookies[name]) {
+        return { name, value: readOnlyCookies[name] };
+      }
+    },
+    set(name, value, options) {
+      setCookie(event, name, value, options);
+    },
+    delete(name) {
+      deleteCookie(event, name);
+    },
+    getAll() {
+      return Object.entries(readOnlyCookies).map(([name, value]) => {
+        return { name, value };
+      });
+    }
+  };
+};
+
+const createLibraryOptions = (
+  event: H3Event<EventHandlerRequest>
+): LibraryOptions => {
+  const cookieStorage = createCookieStorageAdapter(event);
+  const keyValueStorage =
+    createKeyValueStorageFromCookieStorageAdapter(cookieStorage);
+  const tokenProvider = createUserPoolsTokenProvider(
+    amplifyConfig.Auth!,
+    keyValueStorage
+  );
+  const credentialsProvider = createAWSCredentialsAndIdentityIdProvider(
+    amplifyConfig.Auth!,
+    keyValueStorage
+  );
+
+  return {
+    Auth: {
+      tokenProvider,
+      credentialsProvider
+    }
+  };
+};
+
+export const runAmplifyApi = <Result>(
+  // we need the event object to create a context accordingly
+  event: H3Event<EventHandlerRequest>,
+  operation: (
+    contextSpec: AmplifyServer.ContextSpec
+  ) => Result | Promise<Result>
+) => {
+  return runWithAmplifyServerContext<Result>(
+    amplifyConfig,
+    createLibraryOptions(event),
+    operation
+  );
+};
+```
+
+You can then use `runAmplifyApi` function to call Amplify APIs in an isolated server context.
+
+#### Usage Example
+
+Take implementing an API route `GET /api/current-user` , in `server/api/current-user.ts`:
+
+```ts title="server/api/current-user.ts"
+import { getCurrentUser } from 'aws-amplify/auth/server';
+import { runAmplifyApi } from '~/server/utils/amplifyUtils';
+
+export default defineEventHandler(async (event) => {
+  const user = await runAmplifyApi(event, (contextSpec) =>
+    getCurrentUser(contextSpec)
+  );
+
+  return user;
+});
+```
+
+Then you can fetch data from this route, for example, `fetch('http://localhost:3000/api/current-user')`.
+
+## Set Up Server Middleware to Protect Your API Routes
+
+Similar to API routes, the previously added auth middleware are not usable under `/server`, hence extra work is required to set up a auth middleware to protect your routes.
+
+1. If you haven’t already done so, create a `middleware` directory under the `server` directory of your Nuxt project
+2. Create an `amplifyAuthMiddleware.ts` file under the `middleware` directory
+
+This middleware will be executed before a request reach your API route.
+
+Example implementation:
+
+```ts title="middleware/amplifyAuthMiddleware.ts"
```

Comment:
does Nuxt/Vue have a pattern for filenames? filenames earlier in the doc are kebab-case where this is camelCase

Reply:
I don't think so, the snake cases above followed Nuxt documentations, tested using camel case doesn't break them. I can update and unify.